### PR TITLE
Attach experiment receipts to the GH workflow and add build scan links as annotations

### PR DESCRIPTION
**Pull request by jprinet**

### CODE REVIEW

**Comment by jprinet** [comment]
Here is a glimpse of the outcome
https://github.com/jprinet/gradle-enterprise-oss-projects/actions/runs/2258014608

File: .github/actions/gradle/experiment-1/action.yml [hunk 1]
```diff
@@ -70,6 +70,16 @@ runs:
...
+        echo "::set-output name=buildScanFirstBuild::$BUILD_SCAN_1"
+        echo "::set-output name=buildScanSecondBuild::$BUILD_SCAN_2"
+
+        echo "::notice title="Experiment 1 - First Build Scan"::$BUILD_SCAN_1"
+        echo "::notice title="Experiment 1 - Second Build Scan"::$BUILD_SCAN_2"
       shell: bash
+    - name: Archive receipt
+      uses: actions/upload-artifact@v3
+      with:
+        name: receipt-experiment-1
```
Comment (etiennestuder) [inline .github/actions/gradle/experiment-1/action.yml:84]: Please rename to `experiment-1-receipt`

File: .github/actions/gradle/experiment-2/action.yml [hunk 2]
```diff
@@ -70,6 +70,16 @@ runs:
...
+        echo "::set-output name=buildScanFirstBuild::$BUILD_SCAN_1"
+        echo "::set-output name=buildScanSecondBuild::$BUILD_SCAN_2"
+
+        echo "::notice title="Experiment 2 - First Build Scan"::$BUILD_SCAN_1"
+        echo "::notice title="Experiment 2 - Second Build Scan"::$BUILD_SCAN_2"
       shell: bash
+    - name: Archive receipt
+      uses: actions/upload-artifact@v3
+      with:
+        name: receipt-experiment-2
```
Comment (etiennestuder) [inline .github/actions/gradle/experiment-2/action.yml:84]: Please rename to `experiment-2-receipt`

File: .github/actions/gradle/experiment-3/action.yml [hunk 3]
```diff
@@ -70,6 +70,16 @@ runs:
...
+        echo "::set-output name=buildScanFirstBuild::$BUILD_SCAN_1"
+        echo "::set-output name=buildScanSecondBuild::$BUILD_SCAN_2"
+
+        echo "::notice title="Experiment 3 - First Build Scan"::$BUILD_SCAN_1"
+        echo "::notice title="Experiment 3 - Second Build Scan"::$BUILD_SCAN_2"
       shell: bash
+    - name: Archive receipt
+      uses: actions/upload-artifact@v3
+      with:
+        name: receipt-experiment-3
```
Comment (etiennestuder) [inline .github/actions/gradle/experiment-3/action.yml:84]: Please rename to `experiment-3-receipt`

**Review by etiennestuder — CHANGES_REQUESTED** [review]
One request for a renaming in all 3 actions. The PR can then be merged straight to main (after revasing to main first).

**Pull request merged**